internal/provider/resource_tsuru_job.go

```diff
@@ -0,0 +1,324 @@
+// Copyright 2021 tsuru authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package provider
+
+import (
+	"context"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	tsuru_client "github.com/tsuru/go-tsuruclient/pkg/tsuru"
+)
+
+func resourceTsuruJob() *schema.Resource {
+	return &schema.Resource{
+		Description:   "Tsuru Job",
+		CreateContext: resourceTsuruJobCreate,
+		UpdateContext: resourceTsuruJobUpdate,
+		ReadContext:   resourceTsuruJobRead,
+		DeleteContext: resourceTsuruJobDelete,
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(60 * time.Minute),
+			Update: schema.DefaultTimeout(60 * time.Minute),
+			Delete: schema.DefaultTimeout(60 * time.Minute),
+		},
+		Importer: &schema.ResourceImporter{
+			StateContext: resourceTsuruJobImport,
+		},
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:        schema.TypeString,
+				Description: "Job name",
+				Required:    true,
+			},
+			"description": {
+				Type:        schema.TypeString,
+				Description: "Job description",
+				Optional:    true,
+			},
+			"plan": {
+				Type:        schema.TypeString,
+				Description: "Plan",
+				Required:    true,
+			},
+			"team_owner": {
+				Type:        schema.TypeString,
+				Description: "Job owner",
+				Required:    true,
+			},
+			"pool": {
+				Type:        schema.TypeString,
+				Description: "The name of pool",
+				Required:    true,
+			},
+			"tags": {
+				Type:        schema.TypeList,
+				Description: "Tags",
+				Optional:    true,
+				Elem: &schema.Schema{
+					Type: schema.TypeString,
+				},
+			},
+			"metadata": {
+				Type:     schema.TypeList,
+				MaxItems: 1,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"labels": {
+							Type:     schema.TypeMap,
+							Optional: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+						},
+						"annotations": {
+							Type:     schema.TypeMap,
+							Optional: true,
+							Elem:     &schema.Schema{Type: schema.TypeString},
+						},
+					},
+				},
+			},
+
+			"schedule": {
+				Type:        schema.TypeString,
+				Description: "When trigger the job",
```

Comment:
I would write something more informative. 
Like `Cron-like schedule for when the job should be triggered`. 
Can this also be a timestamp? I would assume that by this description ;)